### PR TITLE
[Kubernetes] Add support for Kubernetes v1.31

### PR DESCRIPTION
**Pull request by mauri870**
## Proposed commit message

Kubernetes v1.31.0 was released on 13 Aug 2024.

## Related issues

- Relates https://github.com/elastic/observability-dev/issues/3718

### CODE REVIEW

**Comment by mauri870** [comment]
@elastic/ecosystem The tests are failing for the sqlserver and mysql integrations, but it seems unrelated. The tests for package kubernetes are green.

**Comment by mrodm** [comment]
> @elastic/ecosystem The tests are failing for the sqlserver and mysql integrations, but it seems unrelated. The tests for package kubernetes are green.

Those two packages are failing currently also in main and other PRs, so as you mention those failures are unrelated to the changes on this PR. For instance here:
https://buildkite.com/elastic/integrations/builds/14672

**Review by mrodm — APPROVED** [review]
LGTM

File: packages/kubernetes/_dev/build/docs/README.md [hunk 1]
```diff
@@ -108,7 +108,7 @@ This defaults to `/var/log/kubernetes/kube-apiserver-audit.log`.
 
 ## Compatibility
 
-The Kubernetes package is tested with Kubernetes [1.27.x - 1.30.x] versions
+The Kubernetes package is tested with Kubernetes [1.29.x - 1.31.x] versions
```
Comment (gizas) [inline packages/kubernetes/_dev/build/docs/README.md:111]:
```suggestion
The Kubernetes package is tested with Kubernetes [1.28.x - 1.31.x] versions
```

File: packages/kubernetes/docs/README.md [hunk 2]
```diff
@@ -108,7 +108,7 @@ This defaults to `/var/log/kubernetes/kube-apiserver-audit.log`.
 
 ## Compatibility
 
-The Kubernetes package is tested with Kubernetes [1.27.x - 1.30.x] versions
+The Kubernetes package is tested with Kubernetes [1.29.x - 1.31.x] versions
```
Comment (gizas) [inline packages/kubernetes/docs/README.md:111]:
```suggestion
The Kubernetes package is tested with Kubernetes [1.28.x - 1.31.x] versions
```

File: packages/kubernetes/_dev/build/docs/README.md (hunk 1 above)
Comment (gizas) [reply packages/kubernetes/_dev/build/docs/README.md:111]: According to https://kubernetes.io/releases/ that states that active releases are from 1.28 to 1.31

**Review by gizas — APPROVED** [review]

**Comment by mauri870** [comment]
Can someone with the proper permissions merge this? The CI failures are unrelated. Thanks.

**Comment by mrodm** [comment]
> Can someone with the proper permissions merge this? The CI failures are unrelated. Thanks.

I'll merge this PR @mauri870 .

**Pull request merged**